include/gunrock/algorithms/mst.hxx

```diff
@@ -0,0 +1,302 @@
+/**
+ * @file mst.hxx
+ * @author Annie Robison (amrobison@ucdavis.edu)
+ * @author Muhammad Osama (mosama@ucdavis.edu)
+ * @brief Minimum Spanning Tree algorithm.
+ * @version 0.1
+ * @date 2022-03-17
+ *
+ * @copyright Copyright (c) 2022
+ *
+ */
+#pragma once
+
+#include <gunrock/algorithms/algorithms.hxx>
+
+using vertex_t = int;
+using edge_t = int;
+using weight_t = float;
+
+namespace gunrock {
+namespace mst {
+
+template <typename vertex_t>
+struct param_t {};
+
+template <typename vertex_t, typename weight_t>
+struct result_t {
+  weight_t* mst_weight;
+  result_t(weight_t* _mst_weight) : mst_weight(_mst_weight) {}
+};
+
+template <typename graph_t, typename param_type, typename result_type>
+struct problem_t : gunrock::problem_t<graph_t> {
+  param_type param;
+  result_type result;
+
+  problem_t(graph_t& G,
+            param_type& _param,
+            result_type& _result,
+            std::shared_ptr<cuda::multi_context_t> _context)
+      : gunrock::problem_t<graph_t>(G, _context),
+        param(_param),
+        result(_result) {}
+
+  using vertex_t = typename graph_t::vertex_type;
+  using edge_t = typename graph_t::edge_type;
+  using weight_t = typename graph_t::weight_type;
+
+  graph_t g = this->get_graph();
+  int n_vertices = g.get_number_of_vertices();
+  int n_edges = g.get_number_of_edges();
+
+  thrust::device_vector<vertex_t> roots;
+  thrust::device_vector<vertex_t> new_roots;
+  thrust::device_vector<weight_t> min_weights;
+  thrust::device_vector<int> mst_edges;
+  thrust::device_vector<int> super_vertices;
+  thrust::device_vector<int> min_neighbors;
+  thrust::device_vector<int> mst;
```

Comment:
Can we use appropriate types for these vectors, such as `edge_t` instead of `int`. Wherever it makes sense.